src/test/java/ru/lanwen/verbalregex/RealWorldUnitTest.java

```diff
@@ -149,4 +150,21 @@ public void captureAfterNewLineHasANamedGroup() {
         assertThat(some,
                 equalTo(expression.getText(lineBreak + some + text, captureName)));
     }
+
+    @Test
+    public void missingOptionalCaptureGroupReturnsEmptyStringNotStringContainingNullLiteral() {
+        final VerbalExpression expression = VerbalExpression.regex().
+                startOfLine()
+                .capture("optionalCapture")
+                .oneOf("a", "b")
+                .endCapture()
+                .count(0, 1)
+                .then("c")
+                .endOfLine()
+                .build();
+        final String testString = "c";
+        assertThat(expression, matchesExactly(testString));
+        assertThat(expression.getText("c", "optionalCapture"), not(equalTo("null")));
+        assertThat(expression.getText("c", "optionalCapture"), equalTo(""));
```

Comment:
```suggestion
        assertThat(expression.getText("c", "optionalCapture"), equalTo(""));
```
this one automatically excludes the first one